File: palindrome.js
function isPalindrome(str){
    var pointer1 = 0;
    var pointer2 = str.length - 1;
    while (pointer2 >= pointer1){
        if (str.charAt(pointer2) != str.charAt(pointer1)){
            return false;
        }
        pointer1++;
        pointer2--;
    }
    return true;
}


var test1 = "kayak"
var test2 = "odoo"
var test3 = "solos"
var test4 = "saippuakivikauppias"
var test5 = ""

console.log(isPalindrome(test1));
console.log(isPalindrome(test2));
console.log(isPalindrome(test3));
console.log(isPalindrome(test4));
console.log(isPalindrome(test5));

File: palindrome.py
def isPalindrome(word):
    pointer1 = 0
    pointer2 = len(word) - 1
    while pointer2 >= pointer1:
        if word[pointer1] != word[pointer2]:
            return False
        pointer1 += 1
        pointer2 -= 1
    return True

    


test1 = "kayak"
test2 = "odoo"
test3 = "solos"
test4 = "saippuakivikauppias"
test5 = ""

print(isPalindrome(test1))
print(isPalindrome(test2))
print(isPalindrome(test3))
print(isPalindrome(test4))
print(isPalindrome(test5))